### PR TITLE
[#3660] docs(spark-connector): add multi spark and scala version for spark connector doc

### PR DESCRIPTION
**Pull request by FANNG1**
### What changes were proposed in this pull request?
add multi spark and scala version for spark connector doc

### Why are the changes needed?

Fix: #3600 

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
just document

### CODE REVIEW

File: docs/spark-connector/spark-connector.md [hunk 1]
```diff
@@ -18,13 +18,17 @@ The Gravitino Spark connector leverages the Spark DataSourceV2 interface to faci
 
 ## Requirement
 
-* Spark 3.4
-* Scala 2.12
+* Spark 3.3 or 3.4 or 3.5
+* Scala 2.12 or 2.13
 * JDK 8,11,17
```
Comment (caican00) [inline docs/spark-connector/spark-connector.md:23]: `JDK 8,11,17` should be `JDK 8 or 11 or 17` ?

File: docs/how-to-build.md [hunk 2]
```diff
@@ -88,11 +88,11 @@ This software is licensed under the Apache License version 2."
    with specific module name, like:
 
     ```shell
-    ./gradlew spark-connector:spark-connector-runtime:build
+    ./gradlew spark-connector:spark-runtime-3.4:build -PscalaVersion=2.12
```
Comment (caican00) [inline docs/how-to-build.md:91]: Isn't there a default scala version? For example, if `-PscalaVersion` is not specified, `scala 2.12` is used by default.

Comment (caican00) [reply docs/how-to-build.md:91]: ok. i see it in https://github.com/datastrato/gravitino/pull/3608.

but I still think there should be some extra description here.  for example, to explain the difference between  specifying  -`PscalaVersion` and not specifying.

File: docs/spark-connector/spark-connector.md (hunk 1 above)
Comment (FANNG1) [reply docs/spark-connector/spark-connector.md]: done

File: docs/how-to-build.md (hunk 2 above)
Comment (FANNG1) [reply docs/how-to-build.md:91]: done

File: docs/spark-connector/spark-connector.md [hunk 3]
```diff
@@ -18,13 +18,17 @@ The Gravitino Spark connector leverages the Spark DataSourceV2 interface to faci
...
 
-* Spark 3.4
-* Scala 2.12
-* JDK 8,11,17
+* Spark 3.3 or 3.4 or 3.5
+* Scala 2.12 or 2.13
+* JDK 8 or 11 or 17
+
+:::info
+Gravitino spark connector doesn't support scala 2.13 for Spark3.3.
```
Comment (caican00) [inline docs/spark-connector/spark-connector.md:26]: it seems to be better placed in [how-to-build](https://github.com/datastrato/gravitino/pull/3614/files#diff-ff84168c4ba974b82b38aa13f7e311aaf05a3212a3e56b8af39e83f94b95e90c) doc, WDYT?
to remind users not to use scala2.13 when compiling spark3.3

Comment (FANNG1) [reply docs/spark-connector/spark-connector.md:26]: done

File: docs/how-to-build.md [hunk 4]
```diff
@@ -88,11 +88,17 @@ This software is licensed under the Apache License version 2."
...
     ```
 
    This creates `gravitino-spark-connector-runtime-{sparkVersion}_{scalaVersion}-{version}.jar`
-   under the `spark-connector/spark-connector-runtime/build/libs` directory.
+   under the `spark-connector/v3.4/spark-runtime/build/libs` directory. You could replace `3.4` with 
+   `3.3` or `3.5` to specify different Spark versions, replace `2.12` with `2.13` for different scala 
+   version. The default scala version is `2.12` if not specifying `-PscalaVersion`.
+
+   :::info
+   Gravitino spark connector doesn't support scala 2.13 for Spark3.3.
```
Comment (jerryshao) [inline docs/how-to-build.md:100]: Capitalize `S` for `Scala`

Comment (FANNG1) [reply docs/how-to-build.md]: done

**Review by jerryshao — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Pull request merged**